metadata_layer.py

```diff
@@ -1292,10 +1294,29 @@ def update_contributions(self, _db, edition, metadata_client=None,
                     contributor_data.display_name
                 )
 
+    def filter_recommendations(self, _db):
+        """Filters out  recommendations that don't currently exist in the db
```

Comment:
This is a little misleading. We're filtering out recommended _Identifiers_ that don't exist in the database.

I was wondering if it would also make sense to filter out Identifiers that are not associated with any LicensePool, but you're dealing with that in circulation's RecommendationLane, and I think that makes sense.

Looking up Identifiers one at a time is pretty slow, and you could improve this by doing one lookup for each type of identifier, i.e. gather all the ISBNs and do a query like `_db.query(Identifier).filter(Identifier.type==ISBN).filter(Identifier.identifier.in_(...))`

That will use only one query and tell you which of the ISBNs are actually in the database.